core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/collection/TerminologyResourceCollectionToolingSupport.java

```diff
@@ -0,0 +1,113 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import com.b2international.commons.CompareUtils;
+import com.b2international.commons.exceptions.BadRequestException;
+import com.b2international.snowowl.core.Dependency;
+import com.b2international.snowowl.core.plugin.ClassPathScanner;
+
+/**
+ * @since 9.0
+ */
+public interface TerminologyResourceCollectionToolingSupport {
+
+	/**
+	 * @since 9.0
+	 */
+	public static final class Registry {
+
+		private final Map<String, TerminologyResourceCollectionToolingSupport> collectionToolingSupportImplementations = new HashMap<>();
+
+		public Registry(ClassPathScanner scanner) {
+			scanner.getComponentsByInterface(TerminologyResourceCollectionToolingSupport.class).forEach(this::register);
+		}
+
+		public void register(TerminologyResourceCollectionToolingSupport toolingSupport) {
+			checkArgument(!CompareUtils.isEmpty(toolingSupport.getSupportedChildResourceTypes()), "'%s' resource collection tooling support does not define any valid child resource type.", toolingSupport.getClass().getName());
+			toolingSupport.getSupportedChildResourceTypes().forEach(childResourceType -> {
+				collectionToolingSupportImplementations.put(asKey(toolingSupport.getToolingId(), childResourceType), toolingSupport);
+			});
+		}
+
+		public void unregister(TerminologyResourceCollectionToolingSupport toolingSupport) {
+			toolingSupport.getSupportedChildResourceTypes().forEach(childResourceType -> {
+				collectionToolingSupportImplementations.remove(asKey(toolingSupport.getToolingId(), childResourceType));
+			});
+		}
+
+		public TerminologyResourceCollectionToolingSupport getToolingSupport(String toolingId, String childResourceType) {
+			String key = asKey(toolingId, childResourceType);
+			if (!collectionToolingSupportImplementations.containsKey(key)) {
+				var supportedChildResourceTypes = collectionToolingSupportImplementations.values()
+					.stream()
+					.filter(support -> support.getToolingId().equals(toolingId))
+					.flatMap(support -> support.getSupportedChildResourceTypes().stream())
+					.collect(Collectors.toCollection(TreeSet::new));
+				if (supportedChildResourceTypes.isEmpty()) {
+					throw new BadRequestException("ToolingId '%s' is not supported for resource collections.", toolingId);
+				} else {
+					throw new BadRequestException("ToolingId '%s' and child resource type '%s' combination is not supported for resource collections. ToolingId '%s' supports the following child resource types: '%s'.", toolingId, childResourceType, toolingId, supportedChildResourceTypes);
+				}
+			}
+			return collectionToolingSupportImplementations.get(key);
+		}
+
+		private String asKey(String toolingId, String childResourceType) {
+			return String.join("#", toolingId, childResourceType);
+		}
```

Comment:
Tiny records are also usable as a key! 🙂 It shouldn't matter that much in this particular case, though:

```java
private record Key(String toolingId, String childResourceType) { }
```